src/core/public/chrome/chrome_service.tsx

```diff
@@ -157,6 +157,11 @@ export class ChromeService {
     const recentlyAccessed = await this.recentlyAccessed.start({ http });
     const docTitle = this.docTitle.start({ document: window.document });
 
+    // erase the help menu extension from previous app while switching to a next app
+    application.currentAppId$.subscribe(() => {
+      helpExtension$.next(undefined);
```

Comment:
1. Shouldn't we reset all the Chrome fields then? Badge, Breadcrumbs, etc.
2. It implicitly enforces calling order for `navigation` and `setHelpExtension`. @joshdover should we make this logic explicit? For example, we can move some Chrome params into the `Application` interface?

Reply:
> 1. Shouldn't we reset all the Chrome fields then? Badge, Breadcrumbs, etc.
> 2. It implicitly enforces calling order for `navigation` and `setHelpExtension`. @joshdover should we make this logic explicit? For example, we can move some Chrome params into the `Application` interface?

---

1. Good point, I would also do this on switching an app.
  Especially since I still see couple of places when a page `title` is not changed after navigating to any app, or such a PR's like https://github.com/elastic/kibana/pull/71045 , where `chrome.docTitle.reset()` is manually called on application destroy.
  So doing it once after switching to any app would close other issues.
2. I was also under assumption to set default chrome params explicitly in `navigateToApp`, but discovered that `chrome` service depends on `application` one.
So, the other implementation way would cause significant code changes.

I'm excited to receive @joshdover opinion! 🙂

Reply:
1. Yes, let's go ahead and do those for the chrome items that make sense.
2. I agree this would be much better. I think it would be best to split that change into two phases where we first introduce the new API, separately migrate each usage of the existing imperative APIs, and then remove the imperative APIs. This could be done as a different issue / PR.

Reply:
@joshdover  Should I expand this PR with other chrome items reset, or close it in favor of new API ?

Reply:
> Should I expand this PR with other chrome items reset, or close it in favor of new API ?

We can fix it here quickly and refactor chrome service after. Another complain: https://github.com/elastic/kibana/issues/72671

Reply:
Yes I agree, let's go ahead and do the quick fix now in this PR and refactor later.